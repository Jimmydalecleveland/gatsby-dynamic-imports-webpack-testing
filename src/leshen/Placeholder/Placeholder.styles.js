import styled from '@emotion/styled'
import { css } from '@emotion/core'

export const Placeholder = styled.div(
  ({ theme, color }) => css`
    background-color: ${theme.colors.light.base.hex};
    border-radius: ${theme.radius};
    border: 1px solid ${theme.colors[color].base.hex};
    box-shadow: ${theme.shadow};
    display: flex;
    align-items: stretch;

    * {
      color: ${theme.colors.dark.base.hex};
    }
  `
)

export const IconWrapper = styled.div(
  ({ theme, color }) => css`
    flex: 0 0 auto;
    background-color: ${theme.colors[color].base.hex};
    padding: ${theme.spacing.md} ${theme.spacing.sm};

    svg {
      fill: ${theme.colors[color].base.readable};
    }
  `
)

export const Message = styled.div(
  ({ theme }) => css`
    color: ${theme.colors.dark.base.hex};
    padding: ${theme.spacing.xs} ${theme.spacing.sm};
    text-align: left;

    p {
      margin: 0;
    }
  `
)
