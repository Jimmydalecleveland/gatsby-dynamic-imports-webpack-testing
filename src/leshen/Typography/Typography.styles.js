import styled from '@emotion/styled'
import { css } from '@emotion/core'
import { rem } from 'polished'
import fluidRange from '../utils/fluidRange'

export const Typography = styled.p(({ theme, variant }) => {
  const { headings, subheadings, body } = theme.fonts
  let size
  let lineheight
  let weight

  /* eslint-disable prefer-destructuring */
  // The data is the same in each section. They'd need to be renamed if desctructured and this seems excessive
  switch (variant) {
    case 'h1':
      size = headings.lg.size
      lineheight = headings.lg.lineHeight
      weight = headings.lg.weight
      break
    case 'h2':
      size = headings.md.size
      lineheight = headings.md.lineHeight
      weight = headings.md.weight
      break
    case 'h3':
      size = headings.sm.size
      lineheight = headings.sm.lineHeight
      weight = headings.sm.weight
      break
    case 'h4':
      size = subheadings.lg.size
      lineheight = subheadings.lg.lineHeight
      weight = subheadings.lg.weight
      break
    case 'h5':
      size = subheadings.md.size
      lineheight = subheadings.md.lineHeight
      weight = subheadings.md.weight
      break
    case 'h6':
      size = subheadings.sm.size
      lineheight = subheadings.sm.lineHeight
      weight = subheadings.sm.weight
      break
    case 'feature':
      size = body.lg.size
      lineheight = body.lg.lineHeight
      break
    case 'body':
      size = body.md.size
      lineheight = body.md.lineHeight
      break
    case 'small':
      size = body.sm.size
      lineheight = body.sm.lineHeight
      break
    case 'legal':
      size = body.xxs.size
      lineheight = body.xxs.lineHeight
      break
    default: {
      size = body.md.size
      lineheight = body.md.lineHeight
    }
  }

  return css`
    ${weight && `font-weight: ${weight}`};
    ${Array.isArray(size)
      ? fluidRange('font-size', size)
      : `font-size: ${rem(size)}`};
    ${Array.isArray(lineheight)
      ? fluidRange('line-height', lineheight)
      : `line-height: ${rem(lineheight)}`};
    ${variant === 'legal' && `opacity: 0.8`};

    /* IE10+ CSS styles go here */
    /* IE needs this width to prevent text bleed in flex containers */
    @media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
      width: 100%;
    }
  `
})
