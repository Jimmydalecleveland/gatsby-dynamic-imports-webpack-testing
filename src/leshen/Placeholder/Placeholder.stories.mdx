import { Story, Meta, Preview, Props, Anchor } from '@storybook/addon-docs/blocks'

import Placeholder, { ticketID } from '.'
import Section from '../Section'
import Typography from '../Typography'
import ComponentProgress from '../ComponentProgress'

<Meta title='Components|Placeholder' component={Placeholder} decorators={[(story) => <><ComponentProgress ticketID={ticketID} />{story()}</>]} />

# Placeholder
<Preview>
  <Story name="default">
    <Section>
      <Placeholder>
        <Typography variant="small">Notice</Typography>
        <Typography>This is a quick notification for you to read</Typography>
      </Placeholder>
    </Section>
  </Story>
</Preview>

## Props
<Props of={Placeholder} />

<Preview>
  <Story name="warning">
    <Section>
      <Placeholder type="warning">
        <Typography variant="small">Notice</Typography>
        <Typography>This is a quick notification for you to read</Typography>
      </Placeholder>
    </Section>
  </Story>
</Preview>

<Preview>
  <Story name="error">
    <Section>
      <Placeholder type="error">
        <Typography variant="small">Notice</Typography>
        <Typography>This is a quick notification for you to read</Typography>
      </Placeholder>
    </Section>
  </Story>
</Preview>
