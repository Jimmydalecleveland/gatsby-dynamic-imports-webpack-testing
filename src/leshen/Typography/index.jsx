import React from 'react'
import PropTypes from 'prop-types'

import * as Styled from './Typography.styles'

const Typography = ({ children, rendersAs, variant, ...props }) => {
  let Component = Styled.Typography

  if (rendersAs) {
    Component = Styled.Typography.withComponent(rendersAs)
  } else if (/h\d/.test(variant)) {
    Component = Styled.Typography.withComponent(variant)
  }

  return (
    <Component variant={variant} {...props}>
      {children}
    </Component>
  )
}

Typography.defaultProps = {
  children: null,
  variant: 'body',
  rendersAs: null,
}

Typography.propTypes = {
  children: PropTypes.node,
  rendersAs: PropTypes.string,
  variant: PropTypes.oneOf([
    'h1',
    'h2',
    'h3',
    'h4',
    'h5',
    'h6',
    'legal',
    'feature',
    'body',
    'small',
  ]),
}

export default Typography

export const ticketID = 'CMS-179'
