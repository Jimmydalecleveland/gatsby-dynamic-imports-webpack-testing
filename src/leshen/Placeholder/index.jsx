import React from 'react'
import PropTypes from 'prop-types'
import { NotificationsActive, Error, Warning } from 'emotion-icons/material'

import * as Styled from './Placeholder.styles'

const Placeholder = ({ type, children }) => {
  let Icon = <NotificationsActive />

  if (type === 'warning') {
    Icon = <Warning />
  } else if (type === 'error') {
    Icon = <Error />
  }

  // TODO: A global config should determine what is "production"
  // TODO: should also log in production to some service
  return process.env.NODE_ENV !== 'production' ? (
    <Styled.Placeholder color={type}>
      <Styled.IconWrapper color={type}>{Icon}</Styled.IconWrapper>
      <Styled.Message>{children}</Styled.Message>
    </Styled.Placeholder>
  ) : null
}

Placeholder.defaultProps = {
  type: 'success',
}

Placeholder.propTypes = {
  type: PropTypes.oneOf(['warning', 'error', 'success']),
  children: PropTypes.node.isRequired,
}

export default Placeholder

export const ticketID = 'CMS-171'
