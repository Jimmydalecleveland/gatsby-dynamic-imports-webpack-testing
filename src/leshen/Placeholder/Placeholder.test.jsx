import React from 'react'
import { render } from 'setup-tests'

import Placeholder from '.'
import Typography from '../Typography'

describe('Placeholder', () => {
  test('should render each type', () => {
    const { container } = render(
      <>
        <Placeholder>
          <Typography variant="small">Notice</Typography>
          <Typography>This is a quick notification for you to read</Typography>
        </Placeholder>
        <Placeholder type="warning">
          <Typography variant="small">Notice</Typography>
          <Typography>This is a quick notification for you to read</Typography>
        </Placeholder>
        <Placeholder type="error">
          <Typography variant="small">Notice</Typography>
          <Typography>This is a quick notification for you to read</Typography>
        </Placeholder>
      </>
    )

    expect(container).toMatchSnapshot()
  })

  test('should return nothing when in production', () => {
    // TODO: Does this need to be cleared before tests after it run?
    process.env.NODE_ENV = 'production'
    const { container } = render(
      <Placeholder>
        <Typography variant="small">Notice</Typography>
        <Typography>This is a quick notification for you to read</Typography>
      </Placeholder>
    )

    expect(container).toBeEmpty()
  })
})
