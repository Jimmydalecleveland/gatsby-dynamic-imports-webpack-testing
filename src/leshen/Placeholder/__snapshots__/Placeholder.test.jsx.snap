// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder should render each type 1`] = `
.emotion-8 {
  background-color: #f4f5f6;
  border-radius: 4px;
  border: 1px solid #2de6bc;
  box-shadow: 0 0.3rem 1rem 0 rgba(0,0,0,0.1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-8 * {
  color: #383d43;
}

.emotion-0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  background-color: #2de6bc;
  padding: 24px 16px;
}

.emotion-0 svg {
  fill: #000;
}

.emotion-6 {
  color: #383d43;
  padding: 12px 16px;
  text-align: left;
}

.emotion-6 p {
  margin: 0;
}

.emotion-2 {
  font-size: 0.875rem;
  line-height: 1.375rem;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-2 {
    width: 100%;
  }
}

.emotion-4 {
  font-size: 1rem;
  line-height: 1.375rem;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-4 {
    width: 100%;
  }
}

.emotion-18 {
  background-color: #f4f5f6;
  border-radius: 4px;
  border: 1px solid #ff7928;
  box-shadow: 0 0.3rem 1rem 0 rgba(0,0,0,0.1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-18 * {
  color: #383d43;
}

.emotion-10 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  background-color: #ff7928;
  padding: 24px 16px;
}

.emotion-10 svg {
  fill: #000;
}

.emotion-28 {
  background-color: #f4f5f6;
  border-radius: 4px;
  border: 1px solid #ff3068;
  box-shadow: 0 0.3rem 1rem 0 rgba(0,0,0,0.1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-28 * {
  color: #383d43;
}

.emotion-20 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  background-color: #ff3068;
  padding: 24px 16px;
}

.emotion-20 svg {
  fill: #000;
}

<div>
  <div
    class="emotion-8 emotion-9"
    color="success"
  >
    <div
      class="emotion-0 emotion-1"
      color="success"
    >
      <span>
        NotificationsActive icon Mock
      </span>
    </div>
    <div
      class="emotion-6 emotion-7"
    >
      <p
        class="emotion-2 emotion-3"
      >
        Notice
      </p>
      <p
        class="emotion-4 emotion-3"
      >
        This is a quick notification for you to read
      </p>
    </div>
  </div>
  <div
    class="emotion-18 emotion-9"
    color="warning"
  >
    <div
      class="emotion-10 emotion-1"
      color="warning"
    >
      <span>
        Warning icon Mock
      </span>
    </div>
    <div
      class="emotion-6 emotion-7"
    >
      <p
        class="emotion-2 emotion-3"
      >
        Notice
      </p>
      <p
        class="emotion-4 emotion-3"
      >
        This is a quick notification for you to read
      </p>
    </div>
  </div>
  <div
    class="emotion-28 emotion-9"
    color="error"
  >
    <div
      class="emotion-20 emotion-1"
      color="error"
    >
      <span>
        Error icon Mock
      </span>
    </div>
    <div
      class="emotion-6 emotion-7"
    >
      <p
        class="emotion-2 emotion-3"
      >
        Notice
      </p>
      <p
        class="emotion-4 emotion-3"
      >
        This is a quick notification for you to read
      </p>
    </div>
  </div>
</div>
`;
